cmd/mattermost/commands/server.go

```diff
@@ -69,7 +69,7 @@ func runServer(configStore *config.Store, usedPlatform bool, interruptChan chan
 
 	options := []app.Option{
 		app.ConfigStore(configStore),
-		app.RunJobs,
+		app.RunEssentialJobs,
```

Comment:
What is exactly the purpose of this flag? It seems to me that the only time it's not being set is when other CLI commands are being run.

Can we differentiate based on some other flag, and remove this one altogether?

Reply:
My understanding is that it was added to avoid running jobs during tests and as you said other CLI commands.
As far as naming, I am open to suggestions. I've called it this way because it's what it does, it runs those jobs that are essential to properly run the server (e.g. security jobs).

Reply:
Hmm .. but it also calls `a.srv.runJobs()` which seems to start the job server which starts all the jobs?

I think naming wise, `runJobs` may be the right choice, but I'm 0/5 on that.

What I was saying is that the core of the idea seems to come from the fact of whether we are starting the server as the main service, or via other modes like a CLI command or running a test. The fact that we want to skip the jobs is just one aspect of it, but there can be many. So I was thinking whether there was a separate flag or a way to detect that.

If there was something like

```go
if mainMode {
  // run jobs
}
```

then it makes things much more clear.

Reply:
Okay, I see where the confusion generated now. Obviously it wouldn't make sense to run the other jobs if we don't start those essential to the functioning of the application hence everything goes under the same stricter case.

If we went with "modes" as you suggested I feel like the default should be the main one and the other cases should be adding a special option to deviate from that.

Reply:
Yeah, but I don't want to add any additional flag for that. I initially thought something already exists like that which will let us differentiate. But if there isn't, we can leave it as is.

Reply:
Okay, then if you find it clearer we could still reverse the existing flag and make it a "NoJobs" set by tests and CLI commands and leave the default behaviour of running jobs on startup.

Reply:
Yeah, I think that'll be clearer, but it's beyond the scope of this PR. Fine to be done separately.